packages/itwinui-react/tsconfig.test.json

```
@@ -5,6 +5,7 @@
     "src/core/Table/**/*",
     "src/react-table/*",
     "src/core/Select/**/*",
-    "src/core/LabeledSelect/**/*"
+    "src/core/LabeledSelect/**/*",
+    "src/core/ComboBox/**/*"
```

nit: you can use a pattern like

```suggestion
    "src/**/*.test-types.tsx"
```

---

Actually, with the old pattern, the intention was to test all files in that component path, and not just the type tests.

---

so do we plan to keep adding components one-by-one to this list?

---

Yes, but only for now.

I had actually created a TODO in #1514 (3rd bullet point) as a task to test types in all components. I can create an issue to track that and work on that after all higher priority tasks are complete. After that PR, this include list can just be something like `src/core/**/*`.

---

If you think about it, this PR (and #1774) only adds the `test-types` file, specifically for the purpose of testing the generic types.

If we weren't testing `ComboBox.test.tsx` and `Select.test.tsx` before, why should we do it now? It's equivalent to suddenly testing `Alert.test.tsx`.

Once we migrate to vitest, we can test all our `.test.tsx` files, but for now only `test-types.tsx` files are relevant, so I stand by my original suggestion.

---

> If we weren't testing `ComboBox.test.tsx` and `Select.test.tsx` before, why should we do it now? It's equivalent to suddenly testing `Alert.test.tsx`.

Existing setup of the repo does not catch type errors in tsx files. So, if one complex usage of the `ComboBox` in a unit test in `ComboBox.test.tsx` triggers a type error but the component works correctly, the `jest` test still will pass. So, we will not realize that the types of `ComboBox` might need to be investigated/fixed.

> Once we migrate to vitest, we can test all our `.test.tsx` files, but for now only `test-types.tsx` files are relevant, so I stand by my original suggestion.

Sure, we can defer type testing all files to when we migrate to `vitest`, but I don't see why `vitest` is a pre-requisite for type testing all files. To clarify, my suggestion is using `tsc` to spot type errors in existing files and creating new type test files that use `jest` or `vitest` helper functions to test the types.

> If you think about it, this PR (and #1774) only adds the `test-types` file, specifically for the purpose of testing the generic types.

Sure, we can test only the `types-test.tsx` files for now. The reason I am suggesting including component paths directly is so we can incrementally proceed to our goal of eventually including all component paths in this tsc test. The other option is to do it all at once later and then fix the type errors that occur.

---

These two points are directly connected:

> we can defer type testing all files to when we migrate to vitest, but I don't see why vitest is a pre-requisite for type testing all files.

> The other option is to do it all at once later and then fix the type errors that occur.

We will need to make changes in lots of test files anyway, so fixing type errors can be handled as part of the same migration.

And just to clarify, I'm suggesting we start doing this migration ***now***. There is no need for a "temporary" or "incremental" stage. It is time to do it the right way.

---

If we are migrating to `vitest` now, then yes, fixing type errors and then updating the `tsconfig.test.json`'s `include` can come in that PR.

For this PR, we could add one more component to `include` list since anyways, the test fixes are going to happen soon, and thus that list will be shortened. Or we could go ahead with your [suggestion](https://github.com/iTwin/iTwinUI/pull/1780#discussion_r1449235703). I'm okay either way 👍